pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -220,6 +231,13 @@ func (r *KsvcReconciler) Reconcile() (*knservingv1.ServiceStatus, error) {
 }
 
 func semanticEquals(desiredService, service *knservingv1.Service) bool {
+	for ksvcAnnotationKey := range managedKsvcAnnotations {
+		if desiredValue, ok := desiredService.ObjectMeta.Annotations[ksvcAnnotationKey]; ok {
+			if service.ObjectMeta.Annotations[ksvcAnnotationKey] != desiredValue {
```

Comment:
I think we need additional check here as the annotation may not exist on the current service or it exists but you removed it from desired.

Reply:
@yuzisun that's correct. I modified the function and tested it with several cases. I think all cases work as expected

- add new annotation to isvc -> annotation is added to ksvc
- modify existing annotation in isvc -> annotation is modified in ksvc
- remove existing annotation from isvc -> annotation is removed in ksvc
- manually add / modify / remove annotation in ksvc -> reconciled